ui/components/multichain/pages/review-permissions-page/review-permissions-page.tsx

```diff
@@ -111,22 +121,45 @@ export const ReviewPermissions = () => {
     dispatch(hidePermittedNetworkToast());
   };
 
-  const networkConfigurations = useSelector(getNetworkConfigurationsByChainId);
+  const networkConfigurationsByCaipChainId = useSelector(
+    getAllNetworkConfigurationsByCaipChainId,
+  );
+
   const [nonTestNetworks, testNetworks] = useMemo(
     () =>
-      Object.entries(networkConfigurations).reduce(
+      Object.entries(networkConfigurationsByCaipChainId).reduce(
         ([nonTestNetworksList, testNetworksList], [chainId, network]) => {
-          const isTest = (TEST_CHAINS as string[]).includes(chainId);
-          (isTest ? testNetworksList : nonTestNetworksList).push(network);
+          const caipChainId = chainId as CaipChainId;
+          const isTestNetwork =
+            CAIP_FORMATTED_EVM_TEST_CHAINS.includes(caipChainId);
+          (isTestNetwork ? testNetworksList : nonTestNetworksList).push({
+            ...network,
+            caipChainId,
+          });
           return [nonTestNetworksList, testNetworksList];
         },
-        [[] as NetworkConfiguration[], [] as NetworkConfiguration[]],
+        [
+          [] as EvmAndMultichainNetworkConfigurationsWithCaipChainId[],
+          [] as EvmAndMultichainNetworkConfigurationsWithCaipChainId[],
+        ],
       ),
-    [networkConfigurations],
+    [networkConfigurationsByCaipChainId],
+  );
+
+  const _connectedChainIds = useSelector((state) =>
```

Comment:
Similar as above: 1) This may not be memoized and 2) what is the difference between this and `connectedChainIds`?

Reply:
I've updated the selector to handle the filtering of the `wallet` scopes and added a comment explaining: https://github.com/MetaMask/metamask-extension/pull/29887/commits/99f44c2eb47584e1415b2592dcd390ef020ecea9